calblueprint/wecaresolar
# Created and Styled Video Cards

## 🖋 PR Description

Added an image field (imageUrl) to the video data property of resource and created two different video cards. The video cards are styled to match the displays of the instruction tab and see all view on Figma.

Notes:

- Thank you @kevintli for teaching me so much about back end and adding the image field!
- The video cards were created using the resource card component which renders the article cards
- In order to implement the article card and two video cards using the resource card component, I used a series of if statements to determine if the resource type was an article or video and if the view was for the instructions tab or see all view
- The video card for the see all view was implemented using the favorite button and a FAB button for the duration
- The duration data isn't implemented on the AirTable, so the duration of the video cards is place holder text instead of accessing the data
- Edited the zIndex property of the navBar to remain above the buttons of the video cards

### 🤳Screenshots (if applicable)
**Instructions Tab View**
<img width="497" alt="Screen Shot 2021-04-10 at 1 41 24 AM" src="https://user-images.githubusercontent.com/57604304/114264677-1ec5dc80-99a1-11eb-9c92-147203691308.png">
**See All View**
<img width="498" alt="Screen Shot 2021-04-10 at 1 41 40 AM" src="https://user-images.githubusercontent.com/57604304/114264679-208fa000-99a1-11eb-9fab-1db456d5cf5d.png">


### 🐞 Bug Report

- The primary bug I ran into was accessing the imageUrl property of the video resources. When using `if type == "Article"`, the property said it was not defined for resources. In order to fix the bug, I switched to `if isVideo(resource.data)` which returns a boolean depending on if the resource is a video or not. 

- Another problem I encountered was attempting to use the CardMedia API from Materials UI in order to add media to the card components. Because the image URLs are stored as links rather than files, I used `img src` to display the media on the cards

### 🧪 Testing

How do we test your feature?

To test the feature, you can view the first video card in the instructions tab and then click on the see all view for videos to view the see all video cards.

cc: @susan-lin
